devenv/dev-dashboards/gen.go

```diff
@@ -0,0 +1,93 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"embed"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+)
+
+//go:generate go run gen.go
```

Comment:
wait, does this work? the generator file itself can self-invoke with `//go:generate`? if so, i've been doing [this stupid shit](https://github.com/grafana/grafana/blob/main/public/app/plugins/igen.go#L3) 🤣😭

Reply:
yeah, it seems to, but it could be a terrible idea 😆. 

>  i've been doing
ah, that's where it is! i was attempting to copy some of your patterns from cue generation, and i couldn't track what magick made `go generate` run your `gen.go` files since this one wasn't working. i've probably seen the `//go:generate` comment before, but forgot about it

Reply:
oof, i'm confused - it sounds like you're both saying it seems to work

> yeah, it seems to

but also that it doesn't?

> what magick made `go generate` run your `gen.go` files since this one wasn't working.

FWIW i'm pretty on the fence about where to actually put the `//go:generate` directives. Like, i've considered putting them all in the root `embed.go` file just because it would mean only a single, pretty obvious place to look. But i've kept them sprinkled about basically because of the output that make produces, indicating it's doing a different generation step.

Not a great reason, and one that would probably be better solved by having the loggers print some output on their own. This is also likely to eventually happen given that relying on may cut out Windows devs, so it probably isn't the best place to be the canonical enumeration of all the generators to run.

Reply:
@sdboyer sorry for the confusion. i didn't know (or forgot) that i needed `//go:generate`, so when i ran `go generate gen.go` it was unclear what i was doing wrong. once i added the comment it started working

Reply:
gotcha. I've now tried it out, too, and it does make a difference - if i move the directive out of `public/app/plugins/igen.go` into the (build-tag-ignored) `public/app/plugins/gen.go` that it invokes, then this no longer works:
```
go generate ./public/app/plugins
```

but this does:

```
go generate ./public/app/plugins/gen.go
```

So a bit of a loss, in that it means `go generate ./...` won't work correctly, but given that we don't use that pattern, have no current plans to, and it would plausibly break things like the fact that wire gen depends on and must be run after cue gen, it seems like not a big loss.